File: Cracking Coding Interview/Through C++/Module-2/Assignment/Task_1.cpp
#include <bits/stdc++.h>

using namespace std;

class Task_1 {
	public:
		// Solve the task-1 assignment
		void solution(int nums[], int size, int target) {
			bool brk = false;
			for (int i = 0; i < size-1; i++) {
				for (int j = 0; j < size; j++) {
					if (nums[i]+nums[j] == target) {
						cout << "[" << i << "," << j << "]" << "\n";
						brk = true;
						break;
					}
				}

				if (brk) {
					break;
				}
			}
		}
};


int main() {
	int size;
	cout << "Enter array size: ";
	cin >> size;

	int nums[size];
	cout << "Enter the elements of the array: ";
	for (int i = 0; i < size; i++) {
		cin >> nums[i];
	}

	int target;
	cout << "Enter the target value: ";
	cin >> target;


	Task_1 obj;
	obj.solution(nums, size, target);

	return 0;
}


// Time complexity O(N^2)
// Space Complexity O(N)

File: Cracking Coding Interview/Through Python/LearningToSolving/src/Course/Module-2/Assignment/Task_1.py
class Task_1:
    def __init__(self, nums, target):
        self.nums = nums
        self.target = target
        self.brk = False

    def solution(self):
        # Solution of the task-1 started here
        print(len(self.nums))
        # for i in range(len(self.nums)-1):
        #     for j in range(i+1, len(self.nums)):
        #         if self.nums[i]+self.nums[j] == self.target:
        #             print('[', end='')
        #             print(i, end=',')
        #             print(j, end=']')
        #             self.brk = True
        #             break
        #     if self.brk:
        #         break


# From here execution should be started
size = int(input('Enter the size of the list: '))
lst = []

print('Enter the elements of the list: ')
for k in range(size):
    elm = int(input())
    lst.append(elm)

goal = int(input('Enter the target number: '))

obj = Task_1(lst, goal)


# Time Complexity: O(n^2)
# Space Complexity: O(n)

